### PR TITLE
docs: Update README to link to new docs

### PR DESCRIPTION
**Pull request by lenny-intel**
**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/device-gpio/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] I am not introducing a breaking change (if you are, flag in conventional commit message with `BREAKING CHANGE:` describing the break)
- [x] I am not introducing a new dependency (add notes below if you are)
- [ ] I have added unit tests for the new feature or bug fix (if not, why?)
- [ ] I have fully tested (add details below) this the new feature or bug fix (if not, why?)
- [ ] I have opened a PR for the related docs change (if not, why?)
  <link to docs PR>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -39,251 +50,6 @@ For docker, please refer to the [Dockerfile] and [Docker Compose Builder] script
 [Dockerfile]: Dockerfile
 [Docker Compose Builder]: https://github.com/edgexfoundry/edgex-compose/tree/main/compose-builder
```
Comment (ejlee3) [inline README.md:51]: do we want to link to the latest release of compose builder instead of main?

Comment (lenny-intel) [reply README.md:51]: We don't have a macro for that from the repo readme, so it has to be main

Comment (lenny-intel) [reply README.md:51]: Note on the Releases we create we do point to the version specific Compose Builder.

**Review by ejlee3 — APPROVED** [review]
LGTM

**Pull request merged**